src/api/routineApi.ts

```diff
@@ -1,107 +1,41 @@
-    likes: 673460,
-    userLikes: false,
-  },
-];
+export enum Routine_Endpoint {
+  USER = 'api/workout/routine/user',
+  ROUTINE = 'api/workout/routine',
+  ROUTINE_EXERCISE = 'api/workout/routine/exercise',
+}
 
-export const fetchRoutines = async (query = ''): Promise<Routine[]> => {
-  await new Promise((resolve) => setTimeout(resolve, 1000));
+export const fetchUserRoutines = async (): Promise<Routine[]> => {
+  const response = await api.get(Routine_Endpoint.USER);
 
-  const filteredRoutines: Routine[] = routines.filter((routine) =>
-    routine.name.toLowerCase().includes(query.toLowerCase()),
-  );
-
-  // Uncomment the line below to trigger an error
-  // throw new Error();
-
-  return filteredRoutines;
+  return response.data;
 };
 
-export const fetchRoutineDetails = async (routineId: string): Promise<Routine> => {
-  await new Promise((resolve) => setTimeout(resolve, 1000));
-
-  const routineDetails: Routine | undefined = routines.find((routine) =>
-    routine.id === routineId
-  );
+export const fetchRoutineWithExercises = async (routineId: string): Promise<RoutineWithExercises> => {
+  const response = await api.get(`${Routine_Endpoint.ROUTINE_EXERCISE}?routineId=${routineId}`);
 
-  if (routineDetails === undefined) {
-    throw new Error();
-  }
-
-  return routineDetails;
+  return response.data;
 };
 
-export const createRoutine = async (data: {name: string, description?: string}): Promise<Routine> => {
-  await new Promise((resolve) => setTimeout(resolve, 1000));
-
-  const newRoutine: Routine = {
-    id: (Math.random()*1000).toString(),
-    name: data.name,
-    description: data.description,
-    likes: Math.floor(Math.random()*100000),
-    userLikes: false
-  };
-  routines.push(newRoutine);
+export const createRoutine = async (data: { name: string; description?: string }): Promise<Routine> => {
+  const response = await api.post(`${Routine_Endpoint.ROUTINE}`, data);
 
-  return newRoutine;
+  return response.data;
 };
 
-export const updateRoutine = async (newRoutine: Routine): Promise<Routine> => {
-  await new Promise((resolve) => setTimeout(resolve, 1000));
+// todo: update with exercises, for now only the name and description is updated
```

Comment:
make sure to add that task to jira board, try to describe it as much as possible, so we dont miss it